keyboards/preonic/keymaps/elisiano/keymap.c

```diff
@@ -0,0 +1,314 @@
+  _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU,  KC_MPLY
+),
+
+/* Adjust (Lower + Raise)
+ * ,-----------------------------------------------------------------------------------.
+ * |  F1  |  F2  |  F3  |  F4  |  F5  |  F6  |  F7  |  F8  |  F9  |  F10 |  F11 |  F12 |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      | Reset|      |      |      |      |      |      |      |      |      |  Del |
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * |      |      |      |Aud on|AudOff|AGnorm|AGswap|Qwerty|Colemk|Dvorak|      |      |
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * |      |Voice-|Voice+|Mus on|MusOff|MidiOn|MidOff| Mouse|      |      |      |      |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      |      |      |             |      |      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_ADJUST] = LAYOUT_preonic_grid(
+  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
+  _______, RESET,   DEBUG,   _______, _______, _______, _______, TERM_ON, TERM_OFF,_______, _______, KC_DEL,
+  _______, _______, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, QWERTY,  COLEMAK, DVORAK,  _______, _______,
+  _______, MUV_DE,  MUV_IN,  MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  MOUSE , _______, _______, _______, _______,
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
+),
+
+
+[_MOUSE] = LAYOUT_preonic_grid(
+  _______, KC_ACL0, KC_ACL1, KC_ACL2, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, KC_BTN3, XXXXXXX, XXXXXXX,
+  _______, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, KC_BTN1, KC_MS_U, KC_BTN2, KC_WH_U,
+  _______, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, QWERTY,  KC_MS_L, KC_MS_D, KC_MS_R, KC_WH_D,
+  _______, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, KC_WH_L, XXXXXXX, KC_WH_R, XXXXXXX,
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
+)
+
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+        case QWERTY:
+          if (record->event.pressed) {
+            set_single_persistent_default_layer(_QWERTY);
+          }
+          return false;
+          break;
+        case EMOJI:
+          if (record->event.pressed) {
+            tap_code16(C(G(KC_SPC)));
+          }
+          return false;
+          break;
+        case MOUSE:
+          if (record->event.pressed) {
+            set_single_persistent_default_layer(_MOUSE);
+          }
+          return false;
+          break;
```

Comment:
I'm not sure that you want this.  Having a `TO` keycode to get there and get back mau be closer to what you want.  

As is, this will stay, if you unplug or reset the keyboard.  If that's fine, then leave as is.  Otherwise, you may want to change this.

Reply:
As I’m not familiar with qmk, I was trying to follow how other things are done. 
So if you know a better way of doing the same please let me know (in details though, because I’m missing context). 

That said, this works. I can switch to the Mouse layer and back to another without issues.

Reply:
@drashna do you have any pointers on how to improve that?
Although I'm curious to understand why you said this will stay as in fact I can switch back to any other adjustment layer (with `LOWER+RISE+<KEY>`).

Reply:
Well, the best answers are in the documentation. 

https://docs.qmk.fm/#/ref_functions
https://docs.qmk.fm/#/feature_advanced_keycodes?id=switching-and-toggling-layers

Reply:
@elisiano: I think what @drashna was getting at here (correct me if I'm wrong) is that by using `set_single_persistent_default_layer(_MOUSE);`, you are changing the default layer and storing that to EEPROM, meaning if the board loses power, that would be your layer when plugging back in, which could be confusing.

You could replace the `MOUSE` keycode with `TO(_MOUSE)` and the `QWERTY` keycode on the `_MOUSE` layer with `TO(_QWERTY)` to do this is a way that doesn't involve the EEPROM.

That said, if this works for you as is, I see no reason to block merging this as it looks good besides what drashna pointed out.

Reply:
^ This is exactly what I meant.  

Also, `TG(_MOUSE)` would work too.

Reply:
I see, thanks @ridingqwerty for the detailed explanation.
@drashna I will update the PR as soon as I get a chance!

Reply:
I think I've addressed this, please let me know if there is anything else I need to look into.